app/V1Module/presenters/ShadowAssignmentsPresenter.php

```diff
@@ -0,0 +1,377 @@
+   * @Param(type="post", name="maxPoints", validation="numericint", description="A maximum of points that user can be awarded")
+   * @Param(type="post", name="sendNotification", required=false, validation="bool", description="If email notification should be sent")
+   * @throws BadRequestException
+   * @throws InvalidArgumentException
+   * @throws NotFoundException
+   */
+  public function actionUpdateDetail(string $id) {
+    $assignment = $this->shadowAssignments->findOrThrow($id);
+
+    $req = $this->getRequest();
+    $version = intval($req->getPost("version"));
+    if ($version !== $assignment->getVersion()) {
+      throw new BadRequestException("The shadow assignment was edited in the meantime and the version has changed. Current version is {$assignment->getVersion()}.");
+    }
+
+    // localized texts cannot be empty
+    if (count($req->getPost("localizedTexts")) == 0) {
+      throw new InvalidArgumentException("No entry for localized texts given.");
+    }
+
+    // old values of some attributes
+    $wasPublic = $assignment->isPublic();
+    $isPublic = filter_var($req->getPost("isPublic"), FILTER_VALIDATE_BOOLEAN);
+    $sendNotification = $req->getPost("sendNotification") ? filter_var($req->getPost("sendNotification"), FILTER_VALIDATE_BOOLEAN) : true;
+
+    $assignment->incrementVersion();
+    $assignment->updatedNow();
+    $assignment->setIsPublic($isPublic);
+    $assignment->setIsBonus(filter_var($req->getPost("isBonus"), FILTER_VALIDATE_BOOLEAN));
+    $assignment->setMaxPoints($req->getPost("maxPoints"));
+
+    if ($sendNotification && $wasPublic === false && $isPublic === true) {
+      // assignment is moving from non-public to public, send notification to students
+      $this->assignmentEmailsSender->assignmentCreated($assignment);
+    }
+
+    // go through localizedTexts and construct database entities
+    $localizedTexts = [];
+    foreach ($req->getPost("localizedTexts") as $localization) {
+      $lang = $localization["locale"];
+
+      if (array_key_exists($lang, $localizedTexts)) {
+        throw new InvalidArgumentException("Duplicate entry for language '$lang' in localizedTexts");
+      }
+
+      // create all new localized texts
+      $externalAssignmentLink = trim(Arrays::get($localization, "link", ""));
+      if ($externalAssignmentLink !== "" && !Validators::isUrl($externalAssignmentLink)) {
+        throw new InvalidArgumentException("External assignment link is not a valid URL");
+      }
+
+      $localized = new LocalizedShadowAssignment(
+        $lang, $localization["name"], $localization["text"],
+        $externalAssignmentLink ?: null
+      );
+
+      $localizedTexts[$lang] = $localized;
+    }
+
+    // make changes to database
+    Localizations::updateCollection($assignment->getLocalizedTexts(), $localizedTexts);
+
+    foreach ($assignment->getLocalizedTexts() as $localizedText) {
+      $this->shadowAssignments->persist($localizedText, false);
+    }
```

Comment:
Quoting @Teyras 
> Because Doctrine doesn't really care what you pass to the `persist` method. The call itself is to let Doctrine know of any new localized texts that might have been created by the previous line.